com.unity.render-pipelines.universal/Runtime/Passes/MainLightShadowCasterPass.cs

```diff
@@ -181,16 +181,17 @@ void RenderMainLightCascadeShadowmap(ref ScriptableRenderContext context, ref Cu
                 CoreUtils.SetKeyword(cmd, ShaderKeywordStrings.MainLightShadowCascades, shadowData.mainLightShadowCascadesCount > 1);
                 CoreUtils.SetKeyword(cmd, ShaderKeywordStrings.SoftShadows, softShadows);
 
-                SetupMainLightShadowReceiverConstants(cmd, shadowLight, softShadows);
+                SetupMainLightShadowReceiverConstants(cmd, shadowLight, shadowData.supportsSoftShadows);
             }
 
             context.ExecuteCommandBuffer(cmd);
             CommandBufferPool.Release(cmd);
         }
 
-        void SetupMainLightShadowReceiverConstants(CommandBuffer cmd, VisibleLight shadowLight, bool softShadows)
+        void SetupMainLightShadowReceiverConstants(CommandBuffer cmd, VisibleLight shadowLight, bool supportsSoftShadows)
         {
             Light light = shadowLight.light;
+            bool softShadows = shadowLight.light.shadows == LightShadows.Soft && supportsSoftShadows;
```

Comment:
What actually uses this new variable?

Reply:
As we need to pass supportsSoftShadows now, instead of passing softShadow and supportsSoftShadows  which would be very conffusing. Now I am passing only supportsSoftShadows which is used later on and recreate softShadows as this is used by properties.